### PR TITLE
Add React to Redux

### PR DESCRIPTION
**Pull request by Ibnballo1**
I did the following in this pull request;

- I make sure that I keep the state immutable in reducers!
- I changed the default state in the books reducer from an empty array to an array with a few books (2 books).
- I imported Redux Provider and the store in the main component of the React app.
- I make sure that list of books displays books from the Redux store by using the useSelector to consume the state.
- I make sure that the list of books displays books from the Redux store by importing the useDispatch method.
- I added functionality to remove books from the store.
- I added functionality to check status.

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -6,15 +6,16 @@ const initialCategory = {
   categories: [],
```
Comment (dasileker) [inline src/redux/categories/categories.js:6]: - [ ] The check status should render a message `Under construction`

File: src/components/Categories.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+import { useDispatch } from 'react-redux/es/exports';
+import { checkStatus } from '../redux/categories/categories';
 
-const Categories = () => (
-  <button type="button">Check Status</button>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  return (
+    <button type="button" onClick={() => dispatch(checkStatus('Status Checked'))}>Check Status</button>
```
Comment (dasileker) [inline src/components/Categories.js:8]: - [ ] The dispatch should render the message stored in the **redux/category** `Under construction`.

Capture d’écran 2022-08-23 173711
Capture d’écran 2022-08-23 173726

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Ibnballo1,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

- Linters are good ✅.
- PR is very Pro ✅.
- The readme file is very detailed✅.


## Required Changes ♻️
_Check the comments under the review._

- Nice work on using redux, however, the category page should display a certain message once click on the `check status` button

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js (hunk 2 above)
Comment (Ibnballo1) [reply src/components/Categories.js:8]: Thanks, @dasileker , please I want to ask if the string should appear when the button is clicked or if it should appear immediately after the page is visited.

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @Ibnballo1              :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for making the requested changes good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**